lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/HttpTrackingURLProvider.java

```diff
@@ -62,11 +62,36 @@ public String getTrackingUrl() {
       return "";
     }
 
+    return getHost() + ":" + getPort();
+  }
+
+  /**
+   * get host of the driver.
+   *
+   * @return
+   */
+  @Override
+  public String getHost() {
+
     try {
-      return InetAddress.getLocalHost().getHostAddress() + ":" + httpServer.getPort();
+      return InetAddress.getLocalHost().getHostAddress();
```

Comment:
Wouldn't it be better to inject and use the `LocalAddressProvider` instead? I would also assign host, port, and URL right in the constructor so that all `.get*()` methods are simple and safe.